pkg/objstore/objstore.go

```diff
@@ -437,6 +437,87 @@ func (b *metricBucket) Name() string {
 	return b.bkt.Name()
 }
 
+type prefixedBucket struct {
```

Comment:
nice idea :+1:

Reply:
Thanks @bwplotka

Reply:
Thanks to @pracucci for this [comment](https://github.com/thanos-io/thanos/issues/1318#issuecomment-660851708) and a detailed explanation over Slack